superset/viz.py

```diff
@@ -2113,17 +2121,13 @@ def process_spatial_data_obj(self, key, df):
         elif spatial.get('type') == 'delimited':
             lon_lat_col = spatial.get('lonlatCol')
             df[key] = df[lon_lat_col].apply(self.parse_coordinates)
-
             if spatial.get('reverseCheckbox'):
-                df[key] = [
-                    tuple(reversed(o)) if isinstance(o, (list, tuple)) else (0, 0)
-                    for o in df[key]
-                ]
+                reverse_latlong()
             del df[lon_lat_col]
         elif spatial.get('type') == 'geohash':
-            latlong = df[spatial.get('geohashCol')].map(geohash.decode)
-            df[key] = list(zip(latlong.apply(lambda x: x[0]),
-                               latlong.apply(lambda x: x[1])))
+            df[key] = df[spatial.get('geohashCol')].map(geohash.decode)
+            if not spatial.get('reverseCheckbox'):
+                reverse_latlong()
```

Comment:
This `if` block is super confusing... can't we fix `geohash.decode` instead?

Reply:
I can wrap it in a lambda that reverses it